src/openrct2/peep/PeepData.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "../entity/Peep.h"
+
+#include <vector>
+
+struct SpriteBounds
+{
+    uint8_t sprite_width;           // 0x00
+    uint8_t sprite_height_negative; // 0x01
+    uint8_t sprite_height_positive; // 0x02
+};
+
+struct PeepAnimation
+{
+    const uint32_t base_image;
+    const SpriteBounds bounds;
+    const std::vector<uint8_t> frame_offsets;
```

Comment:
rather than a vector could we use a span and std::array. You would need the array to be in a separate structure but then at least its all static sized. Unless @ZehMatt you have a better idea?

Reply:
Yes, this is true. Tbh I wanted to move away from separate structures; that's the case right now, which makes the file that much harder to read. However, considering the amount of duplication going on currently, 2/3rds of that might have been noise anyway.

Eventually we'll have to resort to something like std::vector anyway, though, once we load all this from object files.